pkg/services/sqlstore/migrator/dialect.go

```diff
@@ -58,6 +58,7 @@ type Dialect interface {
 	NoOpSQL() string
 
 	IsUniqueConstraintViolation(err error) bool
+	IsPrimaryKeyConstrainViolation(err error) bool
```

Comment:
Q: Do we care or would it make sense to make SQLite `db.isThisError(err, int(sqlite3.ErrConstraintPrimaryKey)) || db.isThisError(err, int(sqlite3.ErrConstraintUnique))`?

Reply:
I thought about this too and I believe that it wouldn't harm but because `IsUniqueConstraintViolation()` is used in several other cases where unique constraints are applied I preferred to be more precise.